lib/bolt/inventory/inventory2.rb

```diff
@@ -136,6 +136,21 @@ def expand_targets(targets)
       end
       private :expand_targets
 
+      def remove_target(current_group, target, desired_group)
```

Comment:
We need to call `target.invalidate_group_cache!` to ensure the config, etc for the target is updated next time it's used. Same as we do for `add_to_group`.

That also implies we need a test somewhere that the resolved values for a target correctly change after removing it from a group.

Reply:
We also need whatever the equivalent of that is for inventory v1.